File: producer-api-go-rest/internal/models/entity.go
package models

import "time"

// CarEntity represents a car entity in the database
type CarEntity struct {
	ID         string     `json:"id"`
	EntityType string     `json:"entity_type"`
	CreatedAt  *time.Time `json:"created_at,omitempty"`
	UpdatedAt  *time.Time `json:"updated_at,omitempty"`
	Data       string     `json:"data"` // JSON string
}

// NewCarEntity creates a new CarEntity with current timestamps
func NewCarEntity(id, entityType, data string) *CarEntity {
	now := time.Now()
	return &CarEntity{
		ID:         id,
		EntityType: entityType,
		CreatedAt:  &now,
		UpdatedAt:  &now,
		Data:       data,
	}
}

File: producer-api-python-rest-lambda-pg/models/entity.py
"""Car entity model."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CarEntity(BaseModel):
    """Car entity model."""
    
    id: str
    entity_type: str
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    data: str  # JSON string
    
    class Config:
        """Pydantic config."""
        json_encoders = {
            datetime: lambda v: v.isoformat() if v else None
        }
